bool is_takeoff
bool is_land
--- 